Core/GDCore/IDE/Events/BehaviorParametersFiller.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2024 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+#include "BehaviorParametersFiller.h"
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "GDCore/Events/Instruction.h"
+#include "GDCore/Extensions/Metadata/MetadataProvider.h"
+#include "GDCore/Extensions/Metadata/ParameterMetadataTools.h"
+#include "GDCore/String.h"
+#include "GDCore/Tools/Log.h"
+
+namespace gd {
+
+bool BehaviorParametersFiller::DoVisitInstruction(gd::Instruction &instruction,
+                                                  bool isCondition) {
+  const auto &metadata = isCondition
+                             ? gd::MetadataProvider::GetConditionMetadata(
+                                   platform, instruction.GetType())
+                             : gd::MetadataProvider::GetActionMetadata(
+                                   platform, instruction.GetType());
+
+  gd::String lastLayerName;
+  gd::ParameterMetadataTools::IterateOverParametersWithIndex(
+      instruction.GetParameters(), metadata.GetParameters(),
+      [&](const gd::ParameterMetadata &parameterMetadata,
+          const gd::Expression &parameterValue, size_t parameterIndex,
+          const gd::String &lastObjectName) {
+        if (parameterMetadata.GetValueTypeMetadata().IsBehavior() &&
+            parameterValue.GetPlainString().length() == 0) {
+
+          auto &expectedBehaviorTypeName =
+              parameterMetadata.GetValueTypeMetadata().GetExtraInfo();
+
+          auto &objectsContainersList =
+              projectScopedContainers.GetObjectsContainersList();
+          auto behaviorNames =
+              objectsContainersList.GetBehaviorsOfObject(lastObjectName, true);
+
+          gd::String *foundBehaviorName = nullptr;
```

Comment:
You could have used an empty string and check if it was empty or not instead of a pointer (which carries the risk of being used on something that was moved/deleted in memory). But I guess this is equivalent here!

Reply:
Like, if the behaviorNames vector is modified for any reason (not even cleared, but something added it in it), there is a chance it's moved in memory and your pointer could be wrong.

Don't risk it: use a string with empty string representing the "null" case :)